### DISCUSSION TITLE
How to disable copyright banner in REPL

### DISCUSSION THREAD

**Discussion by ghost** [cvc5/cvc5#7244] [Q&A]
When no arguments are passed to CVC, it reads input from stdin, and in case the process is started directly from a command line, CVC prints copyright message. Is there a way to disable this behavior? There is `--copyright` option to enable it, but how to disable it? Thanks.

**Comment by 4tXJ7f** [comment]
What exactly is your use case? You could use `--no-interactive`, but then you won't get a copyright message _and_ cvc5 won't print `cvc5>`. Does that work for you?

**Comment by ghost** [comment]
@4tXJ7f

> What exactly is your use case?

We are working on automating a very complex system and part of it includes an AutoHotkey script that opens a Command Prompt window and executes a bunch of things, among others it needs to run cvc5 in order to solve some CSP problem, but it cannot be spawned as a child process (due to how things in our system are currently organized), so we need to rely on AutoHotkey parsing the output of cmd by reading characters from screen and converting them to text. The copyright message complicates our processing x10 times.

In contrast, Z3 does not display any banner by default when `-in` option is passed. I would like cvc5 to have an option<br>like `--no-copyright` or `-hide_banner` (like FFmpeg has) to explicitly disable the banner while keeping the interactive mode.

**Comment by 4tXJ7f** [comment]
It sounds like `--no-interactive` may work for you (despite what its name suggests). With that option, cvc5 should behave the same as `z3 -in`, i.e., you can issue multiple queries and get replies, but there is no other output.

**Comment by ghost** [comment]
Sorry, but we need the `cvc5>` output, as we use it as a delimiter to determine when the previous output is fully printed to the console.

I don't really see what is the hindrance to introduce `--no-copyright` flag (or even better do not print copyright at all unless `--copyright` is provided)? I somewhat understand the cvc5 developers' need for copyright adware, but on the other hand, from a technical point of view, it really decreases user satisfaction. Being forced to see 20+ irrelevant lines of output every time I open cvc5 repl is irritating.

All modern open-source software, and even a lot of proprietary closed-source software, that support CLI also provide an option to hide copyright info, or even don't display the copyright by default. Please consider adding such option to cvc5. Thanks.

**Comment by 4tXJ7f** [comment]
I see, thanks. Does decreasing the verbosity work for you (`-q`)? With a lower verbosity level, we don't print the copyright message but I think that we print all the information that you need. Could you give that a try?

Edit: I'll bring up hiding the copyright message by default in our next meeting.
